src/rest/rest_web_server.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for RESTful HTTP server.
+ */
+
+#ifndef OTBR_REST_REST_WEB_SERVER_HPP_
+#define OTBR_REST_REST_WEB_SERVER_HPP_
+
+#include "rest/connection.hpp"
+
+using otbr::Ncp::ControllerOpenThread;
+using std::chrono::steady_clock;
+
+namespace otbr {
+namespace rest {
+
+class RestWebServer
+{
+public:
+    static RestWebServer *GetRestWebServer(ControllerOpenThread *aNcp);
+
+    otbrError Init();
+
+    otbrError UpdateFdSet(otSysMainloopContext &aMainloop);
+
+    otbrError Process(otSysMainloopContext &aMainloop);
+
+private:
+    // For service
+    RestWebServer(ControllerOpenThread *aNcp);
+
+    Resource mResource;
+
+    // For server configure
+    std::unique_ptr<sockaddr_in> mAddress;
+    int                          mListenFd;
+
+    // For Connection
+    std::unordered_map<int, std::unique_ptr<Connection>> mConnectionSet;
+
+    otbrError UpdateConnections(fd_set &aReadFdSet);
+    otbrError Accept(int aListenFd);
+    bool      SetFdNonblocking(int fd);
+    otbrError InitializeListenFd();
+
+    static const uint32_t                 kMaxServeNum;
+    static const uint32_t                 kPortNumber;
+    static std::unique_ptr<RestWebServer> kRestWebServer;
```

Comment:
```suggestion
    static std::unique_ptr<RestWebServer> sRestWebServer;
```

prefix `k` stands for constants. this is obviously not a constant.